src/tests/Microsoft.PowerFx.Interpreter.Tests/MutationScripts/Remove.txt

```diff
@@ -0,0 +1,14 @@
+﻿>> Set(list, Table({ Name: "One", ID: 1}, { Name: "Two", ID: 2}))
+Table({ID:1,Name:"One"},{ID:2,Name:"Two"})
+
+>> Set(list2, Table(First(list)))
+Table({ID:1,Name:"One"})
+
+>> ForAll(list2, Remove(list, ThisRecord))
+Table(Blank())
+
+>>Set(x, [1,2,3])
+Table({Value:1},{Value:2},{Value:3})
+
+>> ForAll(x,Remove(x, ThisRecord))
```

Comment:
>ForAll(x,Remove(x, ThisRecord)) [](http://example.com/codeflow?start=3&length=31)

This is dangerous - we should not let the collection being 'looped' to be updated. Similar to the exception thrown in C# when we do something like

    var coll = new List<int>() { 1, 2, 3 };
    foreach (var i in coll)
    {
        coll.Add(i);
    }

or

    var coll = new List<int>() { 1, 2, 3 };
    foreach (var i in coll)
    {
        coll.Remove(i);
    }

Reply:
Issue [https://github.com/microsoft/Power-Fx/issues/1696](https://github.com/microsoft/Power-Fx/issues/1696) will address this.

Reply:
In any case, let's not add this test. Tests should always show what is expected in the product - and that is not.